### PR TITLE
feat(ts): add public ts types

### PR DESCRIPTION
**Pull request by ruderngespra**
This adds publicly available and enforced ts interfaces to `runContentfulImport`.

#### TODOS

- Fix ts errors in tests (params partly don't match my docs derived ts expectations) ✅ 
- Expand readme with missing params (see comment) ✅

### CODE REVIEW

**Review by mayakarabula — APPROVED** [review]

File: test/unit/index.test.ts [hunk 1]
```diff
@@ -94,7 +94,9 @@ test('Stops import when default locales does not match', async () => {
             default: true
           }
         ]
-      }
+      },
+      spaceId: 'SPACE_ID',
```
Comment (marcolink) [inline test/unit/index.test.ts:98]: Should we use the same format for both params?

**Review by marcolink — APPROVED** [review]
Great 🎉 
Since we are touching the readme already - should we fix the broken badges?

File: lib/index.ts [hunk 2]
```diff
@@ -32,7 +32,33 @@ function createListrOptions (options) {
...
+  skipContentModel?: boolean,
+  skipLocales?: boolean,
+  skipContentPublishing?: boolean,
+  uploadAssets?: boolean,
+  assetsDirectory?: string,
+  host?: string,
+  proxy?: string,
+  rawProxy?: string,
+  rateLimit?: number,
+  headers?: object,
```
Comment (Vinz93) [inline lib/index.ts:52]: Maybe we can be more specific here
```suggestion
  headers?: [header: string]: string | string[] | undefined,
```

Comment (mayakarabula) [reply lib/index.ts:52]: that's a good idea! the shape is a bit different tho, I will update it
https://github.com/contentful/contentful-import/blob/feat/adjust-types-for-public-interface/lib/utils/headers.ts#L13

**Review by Vinz93 — APPROVED** [review]

**Pull request merged**